src/reader/widgets/ReaderWidget.vue

```diff
@@ -7,6 +7,9 @@
       <div class="reader-container u-flex">
         <Paginator :urn="previous" direction="left" />
         <LoaderBall v-if="gqlLoading" />
+        <ErrorMessage v-else-if="gqlError">
```

Comment:
Our AC for the card talked about handling _errors_ (which we have in the GraphQL Plugin), but our alerts to the user are in one specific case (error in the reader).

I don't think we need to address that here, as likely an error with the main reader will impact functionality enough that we can think of it like Django's 500 handler, but I do think we might consider more patterns in our widgets / re-usable components to handle the varying use cases.

I've pointed out some things that Vue Apollo offers in ![](https://github.trello.services/images/mini-trello-icon.png) [Better leverage vue-apollo](https://trello.com/c/capB481s).

It might also be something we revisit in ![](https://github.trello.services/images/mini-trello-icon.png) [Spike to improve performance](https://trello.com/c/vro9WVik) (where we might have smaller initial queries and then lazily load other data from the dataset using cursors / offsets).